.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,146 @@
+name: Build and test
+
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
+  push:
+    branches:
+      - develop
+      - 'wheel/**'
+  release:
+    types:
+      - created
+      - edited
+
+jobs:
+  pyquil-checks:
+    name: pyquil - Build and test module
+    strategy:
+      matrix:
+        os: ['ubuntu-20.04', 'macos-11', 'windows-2019']
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: '0'
+    - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/* +refs/heads/*:refs/remotes/origin/*
```

Comment:
On Linux, should pull docker images here (see `build_and_test.yml` in pytket-extensions repo).

Reply:
oh, I messed that. Thank you for spotting!

Reply:
I have added them now.